replica/database.cc

```diff
@@ -3184,4 +3184,112 @@ database::on_effective_service_levels_cache_reloaded() {
     co_return;
 }
 
+utils::chunked_vector<uint64_t> compute_random_sorted_ints(uint64_t max_value, uint64_t n_values) {
+    static thread_local std::minstd_rand rng{std::random_device{}()};
+    std::uniform_int_distribution<uint64_t> dist(0, max_value);
+    utils::chunked_vector<uint64_t> chosen;
+    chosen.reserve(n_values);
+    for (size_t i = 0; i < n_values; ++i) {
+        chosen.push_back(dist(rng));
+    }
+    std::ranges::sort(chosen);
+    return chosen;
+}
+
+future<utils::chunked_vector<bytes>> database::sample_data_files(
+    table_id id,
+    uint64_t chunk_size,
+    uint64_t n_chunks
+) {
+    auto ticket = get_units(_sample_data_files_sharded_concurrency_limiter, 1);
+
+    struct state_by_shard {
+        utils::chunked_vector<sstables::sstable_files_snapshot> snapshot;
+        schema_ptr schema;
+        uint64_t local_chunks;
+        uint64_t cross_shard_offset;
+    };
+    std::vector<foreign_ptr<std::unique_ptr<state_by_shard>>> state(smp::count);
+
+    co_await container().invoke_on_all(coroutine::lambda([&] (auto& local_db) -> future<> {
```

Comment:
A sharded<> thing can be passed as a parameter to invoke_on_all and auto-unwrapped.

Reply:
Done.